store/auth.go

```diff
@@ -198,3 +199,67 @@ func RequestPackageAccessMacaroon() (string, error) {
 	}
 	return responseData.Macaroon, nil
 }
+
+// RequestDischargeMacaroon requests a discharge macaroon for an existing macaroon
+func RequestDischargeMacaroon(username, password, macaroon, otp string) (string, error) {
```

Comment:
The terminology here is slightly too generic, since discharging may be done for multiple reasons.

Suggestion:
```
// DischargeAuthCaveat returns a macaroon with the store auth caveat discharged.
func DischargeAuthCaveat(...) (...) {
```